org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -331,7 +337,28 @@ private void createMethodProposalLabel(CompletionProposal methodProposal, Comple
 		setSignature(item, String.valueOf(methodProposal.getSignature()));
 		setDeclarationSignature(item, String.valueOf(methodProposal.getDeclarationSignature()));
 		setName(item, String.valueOf(methodProposal.getName()));
-
+		if (methodProposal.isConstructor() && typeInfo.length() > 0 && item.getData() != null && methodProposal.getRequiredProposals() != null && methodProposal.getRequiredProposals().length > 0) {
+			CompletionProposal requiredProposal = methodProposal.getRequiredProposals()[0];
+			try {
+				Object data = item.getData();
+				if (data instanceof Map) {
+					String uri = ((Map<String, String>) data).get(CompletionResolveHandler.DATA_FIELD_URI);
+					if (uri != null && qualifier != null && requiredProposal.getReplaceEnd() > requiredProposal.getReplaceStart()) {
+						ICompilationUnit unit = JDTUtils.resolveCompilationUnit(uri);
+						if (unit != null) {
+							IDocument document = JsonRpcHelpers.toDocument(unit.getBuffer());
+							String prefix = document.get(requiredProposal.getReplaceStart(), requiredProposal.getReplaceEnd() - requiredProposal.getReplaceStart());
+							if (prefix != null && prefix.indexOf(".") > -1) {
```

Comment:
The new commit works well for the two cases I mentioned previously. One additional concern is performance.

![image](https://user-images.githubusercontent.com/14052197/178988911-a7158a24-2573-44fb-aa3e-3aa672ae42f7.png)

The profiling shows the operation `JDTUtils.resolveCompliationUnit` is not cheap. For all constructor completion proposals, the completion context prefix should be the same, can we calculate it only once and reuse it?